modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs.js

```diff
@@ -315,9 +315,9 @@ export default class Blogs {
 			captionNode.classList.add(CSS_INVISIBLE);
 		}
 
-		window[`${this._config.namespace}coverImageCaptionEditor`].setHTML(
-			STR_BLANK
-		);
+		CKEDITOR.instances[
+			`${this._config.namespace}coverImageCaptionEditor`
+		].setData(STR_BLANK);
```

Comment:
I wonder if this is safe to do. The original setHTML seems to have [handled some edge cases](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp#L302-L325) , beyond happy days setting of data.

Reply:
It looks interesting. Can I create another function that handles the original behavior of setHTML from AUI's component?

Reply:
I'm waiting a definition for it before opening a new PR in blogs team(maybe t-lima?) repo.

Reply:
> Can I create another function that handles the original behavior of setHTML from AUI's component?

@diegonvs Sorry for the late reply!

What AUI component do you mean? If we are adding handling of some general edge cases of updating date, this should be done internally to Balloon Editor component. Now, `setHTML` is a very ugly API. I think what you are doing in this PR is much better, using vanilla CKEditor API. Imo we should:
1. Think if there is a better way to reference BE instance, to update its data. Probably not, but it's something to have in mind.
1. Analyse if there is anything in `setHTML` that we actually need. Are those edge cases still valid? It maybe be worth just letting PR as it is, and fix edge cases if any later.
1. If there is something we need to change, maybe there is a way to implement this with standard API in BE. I'm thinking exposed lifecycle moments  like `onChange`, `onSetData`, `onBeforeLoad`, `onInstanceReady` etc.

Reply:
>maybe t-lima?

Yep, Lima is responsible for Blogs. I'm using [this](https://docs.google.com/spreadsheets/d/1egrstgNEMPf3KQT_KNpaemQ8vFMG5fEHGAmUOhP_UqU/edit#gid=0) to check. There is probably an updated version somewhere in Confluence :sweat_smile:

Reply:
> What AUI component do you mean?

https://github.com/liferay/liferay-portal/blob/3bc9e133e07ebff6c1889ca834c49419570894f3/modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/resources/META-INF/resources/js/alloyeditor.js this component but I understood wrongly, it is defined in the JSP as you mentioned.

**Answering**

1. Think if there is a better way to reference BE instance, to update its data. If it's invoked with tag, probably not. I'm thinking something like propsTransformer, but that may be an overkill. It's something to have in mind.

> If it's invoked with tag

Yes, I'm not creative for it but We can improve by creating a class that receives the namespace and the editorName for handling those things. However, it can be take a long time to implement and test to see side effects. 

> Side note: I never took a look deeply in those things that `ckeditor.jsp` contains and I'm horrified (as We have in brazilian portuguese: kkkkkkkkcrying)

Reply:
2. Analyse if there is anything in setHTML that we actually need. Are those edge cases still valid? It maybe be worth just letting PR as it is, and fix edge cases if any later.

This piece of code:

```js
if (instanceDataReady) {
    ckEditorInstance.setData(data);
}
else {
    instancePendingData = data;
}
```

make me worry about it. I prefer do not touch

Reply:
3. If there is something we need to change, maybe there is a way to implement this with standard API in BE. I'm thinking exposed lifecycle moments like onChange, onSetData, onBeforeLoad, onInstanceReady etc.

I thing those callbacks can be easily replaced with propsTransformer API but it will take more and more time to made these changes. Does make sense adding another story/epic for it?

Reply:
Summarizing changes:

1. Create a class to interact with CKEditor internals and avoid defining things in JSP side
2. Migrate current editor's backend lifecycle to propsTransformer pattern

Does make sense implementing it for this migration? I can spend a time to see if the current way it's worth just to cover this task

Reply:
I can't create just for setHTML and getHTML because it depends in external things defined in JSP context.

For the ones who don't understood the problem yet and for me to make a question to myself:

We have a control for editors in the window:

![Screen Shot 2021-09-06 at 19 08 22](https://user-images.githubusercontent.com/7663513/132262852-ad7bfe36-6f70-47e9-8a99-3595a4322c83.png)
<legend>Where each editor instance is represented in a window property.

and when trying to bring it to our context in a "boy-scouting" way, just creating the necessary functions it's doesn't working 😢  

I spent a time today trying to make things work as I already presented the style but I couldn't yet. Am I in a wrong direction? I need some guidance here /cc @nhpatt @markocikos

~~I was thinking about `extends EventEmitter` as we were doing in the past to have other methods like `dispose`~~

Reply:
> I spent a time today trying to make things work as I already presented the style but I couldn't yet. Am I in a wrong direction?

Which can be translated in: I can be thinking in an "over-engineered" way

Reply:
It is probably best that we have the simplest solution for now, covering edge cases if and when they happen. That would be the solution in this PR, `CKEDITOR.instances[name].setData()`. Some of these cases may no longer be valid, like having to [handle setting data before instance is ready](https://github.com/liferay-frontend/liferay-portal/pull/1397#discussion_r702132029). If we reach a case where this is not enough, we can discuss what option for API layer on top makes the most sense.

Reply:
> Some of these cases may no longer be valid, like having to handle setting data before instance is ready.

Yes, It makes sense. 

> If we reach a case where this is not enough, we can discuss what option for API layer on top makes the most sense.

![omg](https://media.giphy.com/media/l41lVcRXUxjKc7jji/giphy.gif?cid=ecf05e47kql0izo5j4w9ip25xe844yiz1yk62075a74ec7tp&rid=giphy.gif&ct=g)